src/components/RestaurantDetails/RestaurantDetails.js

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable jsx-a11y/img-redundant-alt */
+//Libraries
+import React from 'react';
+
+//Helpers
+import fetchData from '../../api/fetchData';
+import restaurantDetailsReader from './readers/restaurantDetailsReader';
+
+//Constants
+import { INITIAL_STATE } from './constants/RestaurantDetails.general';
+
+//Components
+import Offers from './components/Offers';
+import Filters from './components/Filters';
+import RestaurantInfo from './components/RestaurantInfo';
+import Loader from '../../commonComponents/Loader';
+import ErrorHandler from '../../commonComponents/ErrorHandler';
+
+//CSS
+import './RestaurantDetails.css';
+
+class RestaurantDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = INITIAL_STATE;
+    this.fetchRestaurantDetails = this.fetchRestaurantDetails.bind(this);
+  }
+  //Set state RestaurantDetails
+
+  setRestaurantDetails = (data) => {
+    this.setState({
+      restaurantDetails: data,
+    });
+  };
+
+  setAsLoaded = () => {
+    this.setState({
+      isLoading: false,
+    });
+  };
+
+  fetchRestaurantDetails() {
+    fetchData('restaurantDetails.json')
+      .then((response) => this.setRestaurantDetails(response))
+      .catch((error) => {
+        throw new Error(error);
```

Comment:
If possible show some fallback message instead of just throwing and breaking the page